platform/innovium/sonic-platform-modules-cameo/esc601-32q/modules/at24_smbus.c

```diff
@@ -0,0 +1,847 @@
+/*
+ * at24.c - handle most I2C EEPROMs
```

Comment:
Why not use the at24.c which is part of the Linux kernel?

Reply:
Intel I801 works on SMBus mode.  We are using "24c64"(16bit address) for system eeprom.
in at24.c:
```
	if (!i2c_check_functionality(client->adapter, I2C_FUNC_I2C)) {
		if (chip.flags & AT24_FLAG_ADDR16)
			return -EPFNOSUPPORT;
```
So, we made a modification  in it.

Reply:
Any further comment ?

Reply:
@lguohan to comment.